restapi/src/main/java/io/stargate/web/docsapi/service/DocumentService.java

```diff
@@ -595,7 +594,17 @@ private void validateOpAndValue(String op, JsonNode value, String fieldName) {
           }
           conditions.add(new ListFilterCondition(convertedFieldNamePath, op, valueAsList));
         } else {
-          conditions.add(new SingleFilterCondition(convertedFieldNamePath, op, (String) null));
+          // EQ does not support null values in Cassandra 3.x, thus we would manually move that to
+          // the IN [null]
+          SingleFilterCondition nullFilterCondition =
+              new SingleFilterCondition(convertedFieldNamePath, op, (String) null);
+          if (Objects.equals(nullFilterCondition.getFilterOp(), FilterOp.EQ)) {
+            conditions.add(
+                new ListFilterCondition(
+                    convertedFieldNamePath, FilterOp.IN.rawValue, Collections.singletonList(null)));
```

Comment:
`IN` with `null` elements is not supported by CQL, e.g. in C* 3.11:
```
cqlsh:test> select * from d where p1 = 'a' and val in ('x') allow filtering;

 x | p1 | p2 | val
---+----+----+-----
 1 |  a |    |   x

(1 rows)
cqlsh:test> select * from d where p1 = 'a' and val in (null) allow filtering;
InvalidRequest: Error from server: code=2200 [Invalid query] message="Unsupported null value for column val"
```

So I assume the filtering for this `ListFilterCondition` will filter in memory. However, in this case could we avoid "hiding" the `null` in the list and instead add logic to process `null` values in `EQ` predicates as first class values?.. or at least perhaps create a dedicated condition for `EQ null` cases?

Reply:
Also, as far as I understand, if `property EQ null` is the only query condition, we'll execute a full scan across the collection table, followed by PK queries to fetch specific values for the `property` of each document... That could cause a lot of work on the SG side :thinking:

Reply:
@dimas-b @EricBorczuk OK, from your comments, I actually would think about not doing this, reasoning being:

* We actually hide from the user the fact that we will use the `IN` condition, and yes this could have huge impact if this is the only condition (which you would expect to execute fast). So maybe better would be to keep it forbidden, maybe send better error message back and suggest using `IN (null)` with consequences.
* I tried to understand what @dimas-b was saying about the *and instead add logic to process null values in EQ predicates as first class values*, how would this actually work, I see everywhere that null queering is not supported in C*? Maybe we should do a quick sync today once you are online.

Reply:
> add logic to process null values in EQ predicates as first class values,

By that I meant just code clean-up to make it the handling on `EQ null` conditions explicit in Docs API code. It would not help with C*-level issues.

So, all in all, I tend to agree that it may be best to delay fixing this issue. Hopefully, when we have a more elaborate query planning engine in Doc API we could tackle this again. If we can detect that other more restrictive condition are present, then filtering for null values in memory would not be a problem, I think. It may even be doable in current code, but I think we need to put some more thought into how we communicate query complexities to the end user. I see some code in Docs API that denies queries that hit large volumes of data already (e.g. in [searchRows(...)](https://github.com/stargate/stargate/blob/1d8390b0402b7ea4221db7de71006bbe45afc109/restapi/src/main/java/io/stargate/web/docsapi/service/DocumentService.java#L1193))

Reply:
OK @EricBorczuk I am closing this as not done.